src/v0/destinations/am/transform.js

```diff
@@ -335,14 +335,65 @@ const getDefaultResponseData = (message, rawPayload, evType, groupInfo) => {
   const groups = groupInfo && cloneDeep(groupInfo);
   return { groups, rawPayload };
 };
+
+const userPropertiesPostProcess = (rawPayload) => {
+  const operationList = [
+    '$setOnce',
+    '$add',
+    '$unset',
+    '$append',
+    '$prepend',
+    '$preInsert',
+    '$postInsert',
+    '$remove',
+  ];
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  const { user_properties } = rawPayload;
+  const userPropertiesKeys = Object.keys(user_properties).filter(
+    (key) => !operationList.includes(key),
+  );
+  const duplicatekeys = new Set();
+  // eslint-disable-next-line no-restricted-syntax, guard-for-in
+  for (const key of userPropertiesKeys) {
+    // check if any of the keys are present in the user_properties $setOnce, $add, $unset, $append, $prepend, $preInsert, $postInsert, $remove keys as well as root level
+
+    if (
```

Comment:
how can this scenario happen ?

Reply:
Ideally this should happen in the respective function blocks then when we release this functionality to all customers